bumble/device.py

```diff
@@ -5457,9 +5456,16 @@ def on_connection(
         self_resolvable_address: Optional[hci.Address],
         peer_resolvable_address: Optional[hci.Address],
         role: hci.Role,
-        connection_parameters: Optional[core.ConnectionParameters],
+        connection_interval: int,
+        peripheral_latency: int,
+        supervision_timeout: int,
     ) -> None:
         # Convert all-zeros addresses into None.
+        connection_parameters = Connection.Parameters(
```

Comment:
this code could be moved below, where `Connection.from_pending_connection` is called (is isn't used anywhere else). You could even skip the `connection_parameters=` local assignment, and just call the constructor inline, since it isn't referenced anywhere else.